tutorials/foundry-start-to-end.md

```diff
@@ -0,0 +1,470 @@
+import {ERC20} from "openzeppelin-contracts/contracts/token/ERC20/ERC20.sol";
+
+// This ERC-20 contract mints the specified amount of tokens to the contract creator
+contract MyToken is ERC20 {
+  constructor(uint256 initialSupply) ERC20("MyToken", "MYTOK") {
+    _mint(msg.sender, initialSupply);
+  }
+
+  // An external minting function allows anyone to mint as many tokens as they want
+  function mint(uint256 toMint, address to) external {
+    require(toMint <= 1 ether);
+    _mint(to, toMint);
+  }
+}
+```
+
+As you can see, the OpenZeppelin `ERC20` smart contract is imported by the mapping defined in `remappings.txt`.
+
+The second smart contract will depend on this token contract:  
+
+```
+touch Container.sol
+```
+
+Open the file and add the following to it:
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// Import OpenZeppelin Contract
+import {MyToken} from "./MyToken.sol";
+
+enum ContainerStatus {
+    Unsatisfied,
+    Full,
+    Overflowing
+}
+
+contract Container {
+    MyToken token;
+    uint256 capacity;
+    ContainerStatus public status;
+
+    constructor(MyToken _token, uint256 _capacity) {
+        token = _token;
+        capacity = _capacity;
+        status = ContainerStatus.Unsatisfied;
+    }
+
+    // Updates the status value based on the number of tokens that this contract has
+    function updateStatus() public {
+        address container = address(this);
+        uint256 balance = token.balanceOf(container);
+        if (balance < capacity) {
+            status = ContainerStatus.Unsatisfied;
+        } else if (balance == capacity) {
+            status = ContainerStatus.Full;
+        } else if (_isOverflowing(balance)) {
+            status = ContainerStatus.Overflowing;
+        }
+    }
+
+    // Returns true if the contract should be in an overflowing state, false if otherwise
+    function _isOverflowing(uint256 balance) internal view returns (bool) {
+        return balance > capacity;
+    }
+}
+```
+
+The `Container` smart contract can have its status updated based on how many tokens it holds. It requires a `MyToken` smart contract instance to function, so when we deploy it, we will need logic to ensure that it is deployed with a `MyToken` smart contract.  
+
+## Testing in Foundry {: #testing-in-foundry }
+
+Before we deploy anything to a TestNet or MainNet, however, it's good to test your smart contracts. There are many types of tests:
+
+- **Unit tests** — allow you to test specific parts of a smart contract's functionality. When writing your own smart contracts, it can be a good idea to break functionality into different sections so that it is easier to unit test
+- **Fuzz tests** — allow you to test a smart contract with a wide variety of inputs to check for edge cases
+- **Integration tests** — allow you to test a smart contract when it works in conjunction with other smart contracts, so that you know it works as expected in a deployed environment. 
```

Comment:
```suggestion
- **Integration tests** — allow you to test a smart contract when it works in conjunction with other smart contracts, so that you know it works as expected in a deployed environment
```